nestkernel/stimulating_backend_mpi.h

```diff
@@ -0,0 +1,191 @@
+/*
+ *  stimulating_backend_mpi.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef STIMULATING_BACKEND_MPI_H
+#define STIMULATING_BACKEND_MPI_H
+
+#include "stimulating_backend.h"
+#include "nest_types.h"
+#include "nest_time.h"
+#include <set>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <mpi.h>
+
+/* BeginUserDocs: stimulating backend
+
+.. _stimulating_backend_mpi:
+
+Collect data from MPI communication for updating device
+#######################################################
+
+.. admonition:: Availability
+
+   This stimulating backend is only available if NEST was compiled with
+   :ref:`support for MPI <compile-with-mpi>`.
+
+The `mpi` stimulating backend collects data from MPI channels and
+updates stimulating device just before each run. This is useful for
+co-simulation or for receiving data from an external software.
+
+The creation of this MPI communication is based on the function 
+'MPI_Comm_connect' and 'MPI_Comm_disconnect'. The port name is 
+getting from a file for each device with this backend.
+The name of the file need to be named according to the following 
+pattern:
+
+::
+
+   data_path/label/id_device.txt
+
+The properties ``data_path`` and ``data_prefix`` are global kernel properties.
+This path is only used during the ``Prepare`` call. There is not
+functionality for changing connection during the ``Run``.
+If you want to change the connections during the simulation, you need first 
+to close the previous one by using calling ``Cleanup`` of the simulator.
```

Comment:
```suggestion
The ``data_path`` is a global kernel property, while `label` is a
property of the device in question and `id_device` its node ID. 
This path can only be set outside of a `Run` contexts (i.e. not
after ``Prepare()` has been called, but ``Cleanup()`` has not).
```

Reply:
Did some small corrections and added your changes @jougs